pyobvector/client/ob_vec_json_table_client.py

```diff
@@ -0,0 +1,776 @@
+                if col_type == 'DECIMAL':
+                    factory = JsonTableDecimalFactory(10, 0)
+                else:
+                    decimal_pattern = r'DECIMAL\((\d+),\s*(\d+)\)'
+                    decimal_matches = re.findall(decimal_pattern, col_type)
+                    x, y = decimal_matches[0]
+                    factory = JsonTableDecimalFactory(int(x), int(y))
+                model = factory.get_json_table_decimal_type()
+                return model
+            raise ValueError(f"Invalid column type string: {col_type}")
+
+        def reflect(self, engine: Engine):
+            self.meta_cache = {}
+            with engine.connect() as conn:
+                with conn.begin():
+                    stmt = select(ObVecJsonTableClient.JsonTableMetaTBL).filter(
+                        ObVecJsonTableClient.JsonTableMetaTBL.user_id == self.user_id
+                    )
+                    res = conn.execute(stmt)
+                    for r in res:
+                        if r[1] not in self.meta_cache:
+                            self.meta_cache[r[1]] = []
+                        self.meta_cache[r[1]].append({
+                            'jcol_id': r[2],
+                            'jcol_name': r[3],
+                            'jcol_type': r[4],
+                            'jcol_nullable': bool(r[5]),
+                            'jcol_has_default': bool(r[6]),
+                            'jcol_default': (
+                                r[7]['default']
+                                if isinstance(r[7], dict) else
+                                json.loads(r[7])['default']
+                            ),
+                            'jcol_model': ObVecJsonTableClient.JsonTableMetadata._parse_col_type(r[4])
+                        })
+                    for k, _ in self.meta_cache.items():
+                        self.meta_cache[k].sort(key=lambda x: x['jcol_id'])
+
+                    for k, v in self.meta_cache.items():
+                        logger.info(f"LOAD TABLE --- {k}: {v}")
+
+
+    def __init__(
+        self,
+        user_id: int,
+        uri: str = "127.0.0.1:2881",
+        user: str = "root@test",
+        password: str = "",
+        db_name: str = "test",
+        **kwargs,
+    ):
+        super().__init__(uri, user, password, db_name, **kwargs)
+        self.Base.metadata.create_all(self.engine)
+        self.user_id = user_id
+        self.jmetadata = ObVecJsonTableClient.JsonTableMetadata(self.user_id)
+        self.jmetadata.reflect(self.engine)
+
+    def _reset(self):
+        # Only for test
+        self.perform_raw_text_sql(f"TRUNCATE TABLE {JSON_TABLE_DATA_TABLE_NAME}")
+        self.perform_raw_text_sql(f"TRUNCATE TABLE {JSON_TABLE_META_TABLE_NAME}")
+        self.jmetadata = ObVecJsonTableClient.JsonTableMetadata(self.user_id)
+    
+    def refresh_metadata(self):
+        self.jmetadata.reflect(self.engine)
+
+    def perform_json_table_sql(self, sql: str):
```

Comment:
Recommend to add type annotations for function return value

Reply:
done